src/wix/WixToolset.Core.Burn/Bundles/BurnCommon.cs

```diff
@@ -78,8 +78,8 @@ internal abstract class BurnCommon : IDisposable
         protected const UInt32 BURN_SECTION_OFFSET_FORMAT = 40;
         protected const UInt32 BURN_SECTION_OFFSET_COUNT = 44;
         protected const UInt32 BURN_SECTION_OFFSET_UXSIZE = 48;
-        protected const UInt32 BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE = 52;
-        protected const UInt32 BURN_SECTION_SIZE = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE + 4; // last field + sizeof(DWORD)
+        protected const UInt32 BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE0 = 52;
+        protected const UInt32 BURN_SECTION_MIN_SIZE = BURN_SECTION_OFFSET_UXSIZE + 4; // last field is the UX container, as containers could be authored external. + sizeof(DWORD)
```

Comment:
I'm missing something here. We were predefining the `.wixburn` section in src\burn\stub\StubSection.cpp with all the necessary fields. You didn't touch that file. It was a constant size. We can just write as much data as we want into that section?

Reply:
We didn't move any of the fields in that section, so no need to change it.
The dwContainerCount field had a constant value of 2 (UX + WixAttachedContainer).
The nice thing about the fix is that burn was behaving correctly, only the linking process had to be fixed

Reply:
> The dwContainerCount field had a constant value of 2 (UX + WixAttachedContainer).

Not really relevant to my comment, but if there isn't an attached container, it would be 1 not 2.

> We didn't move any of the fields in that section, so no need to change it.

You're not moving fields but you are adding more fields to the end. There used to be one DWORD at the end for the attached container's size, but now there is a DWORD field for each attached container's size. If this does actually work where the binder can extend the `.wixburn` section as much as it wants, then we should simplify src\burn\stub\StubSection.cpp so it doesn't look like it needs to stay in sync with the binder.

Reply:
If you look at the [PE file format](https://docs.microsoft.com/en-us/windows/win32/debug/pe-format), there's more work we should be doing if our section is not a constant size. We have assumptions in our code that are no longer valid, like the fact that the section would always fit in the smallest possible space. The attached container information is already stored in Burn's manifest, can we stop storing the number of attached containers and their sizes in our section header and require parsing the manifest for that information? This current implementation is relying on the fact that the linker is padding the section with extra bytes, and that we are not checking whether we're reading past the end of our section.

Reply:
Perhaps the easier way would be to limit attached containers number so it doesn't overflow the section size.
Say 10 containers should be way over the top since a x86 executable is bound to be less than 4GB, so 2 containers should fit almost all scenarios

Reply:
I wouldn't want to spend my time putting in an artificial limit, but if you think that's easier then that's fine. If you do go that route then I would prefer that the Burn runtime code uses the section size as the limit instead of hardcoding something.

As far as what that limit should be, here's some things to think about.
* We support x64 and ARM64 in v4. I don't think 4GB is the limit for those executables. Also, people might need more containers because of the number of files (max 65535 per cab) instead of the size.
* The minimum size for the section seems to be 512 bytes. If so, we could support 115 ((512 - 52)/4) attached containers without increasing the size of the executable.